src/hypofuzz/dashboard/test.py

```diff
@@ -47,14 +47,15 @@ def __post_init__(self) -> None:
         reports_by_worker = self.reports_by_worker
         self.reports_by_worker = {}
 
-        # TODO: use k-way merge for nlog(k)) performance, since reports_by_worker
+        # use k-way merge for nlog(k) performance, since reports_by_worker
         # is already sorted.
-        # This sorting won't matter for correctness (once we correctly insert
-        # out-of-order reports), but it will for performance, by minimizing the
-        # number of .bisect calls in the worker reports list.
-        for report in sorted(
-            itertools.chain.from_iterable(reports_by_worker.values()),
-            key=lambda r: r.elapsed_time,
+        #
+        # This sorting doesn't matter for correctness (once we correctly insert
+        # out-of-order reports), but it does for performance, by minimizing the
+        # number of .bisect calls in the worker reports list. (I have not profiled
+        # this).
+        for report in k_way_merge(
+            list(reports_by_worker.values()), key=lambda r: r.timestamp
```

Comment:
always sort by n_inputs or elapsed time, never by timestamp - in the unlikely event they disagree, it's probably because the clock was adjusted.

Reply:
thanks for continuing to remind me about this, it's not a conscious decision on my end, I just default to "timestamp" mentally as the sorting attr